src/commands/dev/config/push.rs

```diff
@@ -0,0 +1,348 @@
+        local_config_string,
+    );
+
+    let secrets_ref: HashMap<&str, &str> = secrets
+        .iter()
+        .map(|(k, v)| (k.as_str(), v.as_str()))
+        .collect();
+
+    if agree_to_update {
+        vault
+            .update_secret(
+                vault_token,
+                &vault_secret_annotation.secret_path,
+                &secrets_ref,
+            )
+            .await?;
+    }
+
+    Ok(())
+}
+
+fn has_diff(old_secret_config: &str, new_secret_config: &str) -> bool {
+    let changeset = TextDiff::from_lines(old_secret_config, new_secret_config);
+
+    changeset
+        .iter_all_changes()
+        .any(|change| matches!(change.tag(), ChangeTag::Delete | ChangeTag::Insert))
+}
+
+async fn select_config(
+    available_config: &HashMap<String, VaultSecretAnnotation>,
+) -> (String, VaultSecretAnnotation) {
+    let selection = Select::with_theme(&ColorfulTheme::default())
+        .items(
+            &available_config
+                .iter()
+                .map(|(config_path, annotation)| {
+                    format!(
+                        "{} \t {}::{}/{}#{}",
+                        remove_parent_directories(config_path),
+                        annotation.source,
+                        annotation.engine,
+                        annotation.secret_path,
+                        annotation.secret_name
+                    )
+                })
+                .collect::<Vec<String>>(),
+        )
+        .with_prompt("Which one do you like to push the changes?")
+        .default(0)
+        .report(false)
+        .interact_opt()
+        .unwrap();
+
+    // Clear the config file count line:
+    println!("\x1B[1A\x1B[K");
+
+    return match selection {
+        Some(index) => {
+            let (config_path, annotation) = available_config.iter().nth(index).unwrap();
+            (config_path.clone(), annotation.clone())
+        }
+        None => {
+            println!("No selection made, exiting...");
+            std::process::exit(0);
+        }
+    };
+}
+
+fn get_available_files() -> Result<Vec<PathBuf>, CliError> {
+    let current_path = current_dir()?;
+    let available_files = get_dev_config_files(&current_path);
+
+    Ok(available_files)
+}
+
+fn filter_config_with_secret_annotations(
+    available_files: Vec<PathBuf>,
+) -> Result<HashMap<String, VaultSecretAnnotation>, CliError> {
+    let mut filtered_annotations: HashMap<String, VaultSecretAnnotation> = HashMap::new();
+
+    for file in available_files {
+        let file_contents = std::fs::read_to_string(file.clone())?;
+        let annotations = read_vault_annotation(&file_contents);
+
+        for annotation in annotations {
+            if annotation.key == "wukong.mindvalley.dev/config-secrets-location"
+                && annotation.source == "vault"
+                && annotation.engine == "secret"
+            {
+                filtered_annotations.insert(file.to_string_lossy().to_string(), annotation);
```

Comment:
There is a bug here, one file could have multiple annotations, but this will only return the last annotation

Reply:
since the possibility of this is small atm, we can address this in another PR later.